### ISSUE TITLE
Add support for gss-tsig

### ISSUE THREAD

**Issue by tokcum** [NLnetLabs/domain#378]
Hi,

I'm evaluating a project where an IT management system (implemented in rust) should add  / update / delete DNS records on a Windows DNS Server. The DNS zones only allow secure writes. Windows DNS Server does not support TSIG but requires GSS-TSIG, i.e. GSS-API has to be used for authentication. HMAC-MD5 is not supported.

These days I'm doing some research to identify the building blocks already available and came across your crate. Is GSS-TSIG support something you would like to see in your crate or would it be preferable to create a new crate for this? What do you think about the difficulty of this to implement?

Thank you for your suggestions and advice in advance.

Tobias

**Comment by partim** [comment]
I think we’d prefer for it to be in a separate crate. I looked at it back when I implemented TSIG for _domain_ and opted against supporting it because it all seemed very complex. Given that it is a relatively niche feature, it is probably better to keep it separate.

That said, we would certainly want you to be able to build GSS-API support on top of our TSIG implementation. I’m not sure that is currently possible but we’re happy to make the necessary changes.
